Unity-MCP-Plugin/Assets/root/Editor/Scripts/API/Tool/TestRunner/TestResultCollector.cs

```diff
@@ -214,67 +219,49 @@ void OnLogMessageReceived(string condition, string stackTrace, LogType type)
             }
         }
 
-        string FormatTestResults(bool includeMessage, bool includeMessageStacktrace, bool includeLogs, bool includeLogsStacktrace)
+        TestRunResponse CreateStructuredResponse(bool includePassingTests, bool includeMessage, bool includeMessageStacktrace, bool includeLogs, bool includeLogsStacktrace)
         {
             var results = GetResults();
             var summary = GetSummary();
             var logs = GetLogs();
 
-            var output = new StringBuilder();
-            output.AppendLine("[Success] Test execution completed.");
-            output.AppendLine();
-
-            // Summary
-            output.AppendLine("=== TEST SUMMARY ===");
-            output.AppendLine($"Status: {summary.Status}");
-            output.AppendLine($"Total: {summary.TotalTests}");
-            output.AppendLine($"Passed: {summary.PassedTests}");
-            output.AppendLine($"Failed: {summary.FailedTests}");
-            output.AppendLine($"Skipped: {summary.SkippedTests}");
-            output.AppendLine($"Duration: {summary.Duration:hh\\:mm\\:ss\\.fff}");
-            output.AppendLine();
-
-            // Individual test results
-            if (results.Any())
+            var response = new TestRunResponse
+            {
+                Summary = summary,
+                Results = new List<TestResultData>()
+            };
+
+            // Filter test results based on includePassingTests, includeMessage and includeMessageStacktrace
+            foreach (var result in results)
             {
-                output.AppendLine("=== TEST RESULTS ===");
-                foreach (var result in results)
+                // Skip passing tests if includePassingTests is false
+                if (!includePassingTests && result.Status == "Passed")
```

Comment:
Lets update `Status` (`result.Status`) to Enum. It would safer and will prevent any string comparison issues such as different lower/upper case, also works faster.

Reply:
Done.